lib/index.js

```diff
@@ -101,81 +41,248 @@ const scrollOptions = {
      this.rotate = config.rotate
      this.flip = config.flip
      this.font = getFont(config.font)
+     this.transform = config.transform
+     this.reverseDisplayOrdering = config.reverseDisplayOrdering
+   }
+   
+   setController (requested = 0) {
+     if (requested < 0 || requested >= this.controllerCount) { 
+       throw new Error(`Requested controller ${requested} is not configured.`)
+     }
+     this.activeController = requested
    }
 
    async displayTest (controller, enable) {
-     setController(this, controller)
-     await shiftOut(this, Registers.DisplayTest, enable ? on : off)
+     if (undefined === enable) {
+       enable = controller;
+       for (controller = 0; controller < this.controllerCount; controller++) {
+         this.setController(controller)
+         await this.shiftOut(Registers.DisplayTest, enable ? on : off)
+       }
+     } else {
+       this.setController(controller)
+       await this.shiftOut(Registers.DisplayTest, enable ? on : off)
+     }
    }
- 
+   
    async reset (controller) {
-     setController(this, controller)
      const zeroArray = new Array(8).fill(0)
-     await shiftOut(this, Registers.DecodeMode, encodeByte(zeroArray))
-     await shiftOut(this, Registers.ScanLimit, 7)
-     await shiftOut(this, Registers.Shutdown, on)
-     await this.displayTest(controller, false)
-     await this.clear(controller)
+     if (undefined === controller) {
+       for (controller = 0; controller < this.controllerCount; controller++) {
+         this.setController(controller)
+         await this.shiftOut(Registers.DecodeMode, 0)
+         await this.shiftOut(Registers.ScanLimit, 7)
+         await this.shiftOut(Registers.Shutdown, on)
+         await this.displayTest(controller, false)
+         await this.clear(controller)
+       }
+     } else {
+       this.setController(controller)
+       await this.shiftOut(Registers.DecodeMode, 0)
+       await this.shiftOut(Registers.ScanLimit, 7)
+       await this.shiftOut(Registers.Shutdown, on)
+       await this.displayTest(controller, false)
+       await this.clear(controller)
+     }
    }
-
+   
    async clear (controller) {
-     setController(this, controller)
-     const matrix = emptyMatrix()
-     await this.set(controller, matrix)
+     const matrix = Max7219.emptyMatrix()
+     if (undefined === controller) {
```

Comment:
same with this one. I think two methods is ultimately more readable.

Reply:
Functions were created.  Please review latest comments.